Java/search/InterpolationSearch.java

```diff
@@ -0,0 +1,55 @@
+import java.util.Scanner;                                                            //importing the packages required for this code
+// JAVA Implementation of recursive Interpolation Search
+public class InterpolationSearch {                                                     
+       int findEleByInterSearching(int arr[], int low, int high, int key) {         // Returns index of key if it is present in arr[low..high], else return -1 
+       if(high >= low && key>=arr[low] && key<=arr[high]) {
+            if(low==high)                                                             //if value of high and low ae same
+            	{
+		    if(arr[low]==key)                                                // value of key compared with element at index low
+            	           return low;
+                        else
+            		   return -1;
+                 }
+           int pos = low + (((high-low)/(arr[high]-arr[low]))*(key-arr[low]));        //determining the position of the key element in an array
+           if(arr[pos] == key)                                                      // If the element key is present at the pos value itself 
+                return pos;
+           if(arr[pos] > key)                                                       // If element key  is smaller than pos value
+                  return findEleByInterSearching(arr, low, pos - 1, key);                // then it can only be present in left subarray 
+           if(arr[pos] < key)                                                       //Else the element key  can only be present 
```

Comment:
Keep the comments above the Code. Not by its side. Use Multi-Line Comments for your purpose

Reply:
Updated the file as required. Kept the comments section above Code and even updated the README file.
Kindly review it  @HarshCasper .